totalRP3/modules/register/characters/register_characteristics.lua

```diff
@@ -312,7 +316,12 @@ local function setConsultDisplay(context)
 	local shownCharacteristics = {};
 	local shownValues = {};
 	for _, attribute in pairs(CHAR_KEYS) do
-		if strtrim(dataTab[attribute] or ""):len() > 0 then
+		if attribute == "RS" then
+			if dataTab[attribute] and dataTab[attribute] > 0 then
+				tinsert(shownCharacteristics, attribute);
+				shownValues[attribute] = RELATIONSHIP_STATUS_DROPDOWN[dataTab[attribute] + 1][1];
```

Comment:
This line is quite confusing. If it's possible to break it down in a more readable code it would make it easier to maintain in the future.
Maybe sprinkle some checks in case the value of `dataTab[attribute]` is not listed in `RELATIONSHIP_STATUS_DROPDOWN` (profiles coming from MSP), as you might be trying to index a `nil` value.